### ISSUE TITLE
how to construct a loss function symbol using python api

### ISSUE THREAD

**Issue by chenjx1005** [dmlc/mxnet#508]
from https://github.com/dmlc/mxnet/blob/master/example/adversary/adversary_generation.ipynb I see how to calculate softmax, gradient in numpy. However, I want a loss function symbol so that i don't need to assign model.backward an out_grads. I try to write code like this but not succeeded:

import mxnet as mx

A = mx.symbol.Variable('A')
B = mx.symbol.Variable('B')
W = mx.symbol.Variable('W')
f = A * W + B

Y = mx.symbol.Variable('Y')
loss = mx.symbol.square((Y - f))

a = mx.nd.array([1])
b = mx.nd.array([3])
w = mx.nd.array([2])
y = mx.nd.array([4])

gw = mx.ndarray.empty(1)

executor = loss.bind(mx.cpu(), args=[y, a, w, b], args_grad={'W':gw})
#executor = loss.simple_bind(mx.cpu())

executor.forward()
print executor.outputs[0].asnumpy()
executor.backward()
print gw.asnumpy() 

./dmlc-core/include/dmlc/logging.h:208: [13:33:25] src/symbol/graph_executor.cc:637: Check failed: (info.ref_count) == (0) Because the last operator is not Loss function, head_gradient is required in calling backward.

**Comment by piiswrong** [comment]
There are two ways:
1. write an operator with cuda kernel under src/operator. You get best performance but need to put in some effort.
2. write an operator in python with numpy like https://github.com/dmlc/mxnet/pull/436. It's fast to develop but may become performance bottleneck

**Comment by chenjx1005** [comment]
thanks for your reply!
